packages/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx

```diff
@@ -135,7 +134,41 @@ export const useGridVirtualScroller = () => {
   const columnsTotalWidth = dimensions.columnsTotalWidth;
   const hasColSpan = useGridSelector(apiRef, gridHasColSpanSelector);
 
-  useResizeObserver(mainRef, () => apiRef.current.resize());
+  const mainRefCallback = React.useCallback((node: HTMLDivElement | null) => {
+    mainRef.current = node;
+
+    if (node) {
+      const initialRect = node.getBoundingClientRect();
+      let lastSize = {
+        width: initialRect.width,
+        height: initialRect.height
+      };
+
+      apiRef.current.publishEvent('resize', lastSize);
```

Comment:
I think this block could possibly be removed, IIUC `ResizeObserver` should fire immediately: https://drafts.csswg.org/resize-observer/#ref-for-element%E2%91%A3

Reply:
The initial value of `lastSize` could be `EMPTY_SIZE` from `useGridDimensions`.